src/components/Personalization/dom-actions/dom/getNonce.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+let nonce;
+
+/**
+ * Returns the nonce if available.
+ * @param {Node} [context=document] defaults to document
+ * @returns {(String|undefined)} the nonce or undefined if not available
+ */
+export default (context = document) => {
+  if (!nonce) {
+    const n = context.querySelector("[nonce]");
+    // NOTE: We're keeping n.getAttribute("nonce") until it is safe to remove:
+    //   ref: https://github.com/whatwg/html/issues/2369#issuecomment-280853946
+    nonce = n && (n.nonce || n.getAttribute("nonce"));
+  }
+  return nonce;
+};
```

Comment:
@Aaronius , I actually implemented it like that initially through memoize, but the functional tests were failing because nonces were being made available in later tests causing them to fail as null being cached. Not sure why cache state was being kept in the memoize between tests though.

Reply:
Are you sure it was a functional test that was failing and not a unit test? I can see why this might fail a later unit test, because I don't think Karma re-initializes modules between unit tests (it would maintain the cached nonce in memory throughout the remainder of tests). If it's a unit test that's failing, then the way I've handled this in the past is by exporting a named constant from the module that's named something like `test_resetCachedNonce` that's a function that just sets the `nonce` variable back to `undefined`. I then run `test_resetCachedNonce` after each of my nonce tests. When we create the production Alloy builds, `test_resetCachedNonce` will get stripped out via Rollup's tree-shaking because nothing else in the source code is using it.

Reply:
Good idea, added

Reply:
Can you add a comment to the testResetCachedNonce explaining that this is only called by test code and gets removed in the final version by tree-shaking? An alternate way of fixing this would be to rename to "injectGetNonce" thereby wrapping the nonce declaration in a function, and then initialize it in index.js. This would require adding dependency injection to more of the Dom-actions stuff though.

Reply:
@jonsnyder , sure I'll add a comment